docs/storage-surface.rst

```diff
@@ -0,0 +1,673 @@
+   >>> bucket.website = {
+   ...     'mainPageSuffix': 'index.html',
+   ...     'notFoundPage': '404.html',
+   ... }
+
+In general, many of these properties are optional and will not need to be
+used (or changed from the defaults).
+
+In addition, a bucket has several read-only properties
+
+.. code-block:: python
+
+   >>> bucket.etag
+   u'CAI='
+   >>> bucket.id
+   u'bucket-name'
+   >>> bucket.metageneration
+   2L
+   >>> bucket.name
+   u'bucket-name'
+   >>> bucket.owner
+   {u'entity': u'project-owners-111111'}
+   >>> bucket.project_number
+   u'111111'
+   >>> bucket.self_link
+   u'https://www.googleapis.com/storage/v1/b/bucket-name'
+   >>> bucket.time_created
+   u'2015-01-01T12:00:00.000Z'
+
+See `buckets`_ specification for more details.
+
+Other data -- namely `access control`_ data -- is associated with buckets,
+but is handled through ``Bucket.acl``.
+
+.. _buckets: https://cloud.google.com/storage/docs/json_api/v1/buckets
+.. _access control: https://cloud.google.com/storage/docs/access-control
+
+.. note::
+  **BREAKING THE FOURTH WALL**: Note that ``storage.buckets.update`` is
+  absent. This doesn't seem necessary to implement given the presence of
+  ``patch``.
+
+Manage Blobs
+------------
+
+Interacting with blobs requires an associated bucket. Either methods on an
+explicit :class:`Bucket <gcloud.storage.bucket.Bucket>` instance can be used
+or the default bucket can be used via methods in the ``storage`` package.
+
+To retrieve a blob with a bucket
+
+.. code-block:: python
+
+   >>> bucket.get_blob('blob-name')
+   <Blob: bucket-name, blob-name>
+
+or from the package via the default bucket
+
+.. code-block:: python
+
+   >>> storage.get_blob('blob-name')
+   <Blob: default-bucket-name, blob-name>
+
+Simply calling ``get_blob`` will not actually retrieve the contents stored
+for the given blob. Instead, it retrieves the metadata associated with
+the blob.
+
+For the remainder of this section we will just illustrate the methods
+on :class:`Bucket <gcloud.storage.bucket.Bucket>`. For each such method,
+an equivalent function is defined in the ``storage`` package and each
+such function relies on the default bucket.
+
+To retrieve multiple blobs in a single request
+
+.. code-block:: python
+
+   >>> blob1, blob2, blob3 = bucket.get_blobs('blob-name1',
+   ...                                        'blob-name2',
+   ...                                        'blob-name3')
+
+This is equivalent to
+
+.. code-block:: python
+
+   >>> with storage.Batch():
+   ...     blob1 = bucket.get_blob('blob-name1')
+   ...     blob2 = bucket.get_blob('blob-name2')
+   ...     blob3 = bucket.get_blob('blob-name3')
+
+however, recall that if ``bucket`` has a ``connection`` explicitly bound
+to it, the ``bucket.get_blob`` requests will be executed individually
+rather than as part of the batch.
+
+To list all blobs in a bucket
+
+.. code-block:: python
+
+   >>> for blob in bucket.list_blobs():
```

Comment:
The absence of arguments to `list_blobs()` is a really bad idea in an application with thousands or millions of blobs. There would be no good defaults to use to `__iter__`.